build-support/docker/travis_ci_py27_ucs2/Dockerfile

```diff
@@ -15,19 +15,23 @@ FROM pantsbuild/centos6:latest
 # interpreter to use when `ci.sh` and `release.sh` set the interpreter constraints for
 # Pants, and thus for the built ./pants.pex. We set $PY to the exact Python 2.7 version we want
 # to ensure the PEX is bootstrapped with UCS 2.
-ARG PYTHON_2_VERSION=2.7.15
+#
+# We also must use pyenv's versions folder, rather than the shims folders, to ensure we use the exact
+# version in the case the base image uses Pyenv to install Python 2.7.
+ARG PYTHON_27_VERSION=2.7.15
 # TODO(7064): remove this yum install line once we update the base Centos6 image to include this dependency.
 RUN yum install sqlite-devel -y
 ENV PYENV_ROOT /pyenv-docker-build
+ENV PYENV_BIN ${PYENV_ROOT}/bin/pyenv
 RUN mkdir ${PYENV_ROOT}
 RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
```

Comment:
Won't this fail if the centos6 base image has already written to `/pyenv-docker-build`? I think temporarily it might make sense just to change that directory. Also note that `ARG` gets set in the environment so if we did `ARG PYENV_ROOT /pyenv-docker-build` we could change what that directory is in our `.travis.yml`, which seems more flexible.

Reply:
Great catch!! I realized thanks to this comment that we can simply rename `PYENV_ROOT` to `/pyenv-docker-build-ucs2` and everything will work out from there. There's no reason we can't have two Pyenvs installed on the image. Because we set `PY` and `PEX_PYTHON_PATH`, there will be no issues with ambiguity on which interpreter to use.